app/controllers/application_controller.rb

```diff
@@ -39,116 +39,81 @@ def routing_error
-
-  def set_tags_and_extra_context
-    Thread.current['request_id'] = request.uuid
-    Thread.current['additional_request_attributes'] = {
-      'remote_ip' => request.remote_ip,
-      'user_agent' => request.user_agent
-    }
-    Raven.extra_context(request_uuid: request.uuid)
-    Raven.user_context(user_context) if current_user
-    Raven.tags_context(tags_context)
+  def set_app_info_headers
+    headers['X-Git-SHA']           = AppInfo::GIT_REVISION
+    headers['X-GitHub-Repository'] = AppInfo::GITHUB_REPO
   end
 
-  def user_context
-    {
-      uuid: current_user&.uuid,
-      authn_context: current_user&.authn_context,
-      loa: current_user&.loa,
-      mhv_icn: current_user&.mhv_icn
-    }
+  def handle_error(error)
+    Common::Client::ErrorHandler.handle(error)
   end
 
-  def tags_context
-    { controller_name: controller_name }.tap do |tags|
-      if current_user.present?
-        tags[:sign_in_method] = current_user.identity.sign_in[:service_name]
-        # account_type is filtered by sentry, becasue in other contexts it refers to a bank account type
-        tags[:sign_in_acct_type] = current_user.identity.sign_in[:account_type]
-      else
-        tags[:sign_in_method] = 'not-signed-in'
-      end
-    end
+  def render_error(exc)
+    render json: { errors: exc.errors }, status: exc.status
   end
 
-  def set_app_info_headers
-    headers['X-GitHub-Repository'] = 'https://github.com/department-of-veterans-affairs/vets-api'
-    headers['X-Git-SHA'] = AppInfo::GIT_REVISION
+  # @note Later rescue_from statements take precedence and so we must take care to respect order
+  rescue_from StandardError do |error|
+    exc = Common::Exceptions::InternalServerError.new(error)
+    handle_error(exc)
+    render json: { errors: exc.errors }, status: exc.status
+  end
+
+  rescue_from Common::Exceptions::BaseError do |exc|
+    set_empty_auth_header if exc.is_a?(Common::Exceptions::Unauthorized)
+    handle_error(exc)
+    render_error(exc)
+  end
+
+  rescue_from ActionController::ParameterMissing do |error|
+    exc = Common::Exceptions::ParameterMissing.new(error.param)
+    handle_error(exc)
+    render_error(exc)
+  end
+
+  rescue_from ActionController::UnknownFormat do |error|
+    exc = Common::Exceptions::UnknownFormat.new(error.param)
+    handle_error(exc)
+    render_error(exc)
+  end
+
+  rescue_from Breakers::OutageException do |error|
+    exc = Common::Exceptions::ServiceOutage.new(error)
+    handle_error(exc)
+    render_error(exc)
+  end
+
+  rescue_from Common::Client::Errors::ClientError do |error|
+    exc = Common::Exceptions::ServiceOutage.new(error, detail: 'Backend Service Outage')
+    handle_error(exc)
+    render_error(exc)
+  end
+
+  rescue_from Pundit::NotAuthorizedError do |error|
+    exc = Common::Exceptions::Forbidden.new(cause: error, detail: 'User does not have access to the requested resource')
+    handle_error(exc)
+    render_error(exc)
+  end
+
```

Comment:
In this case, we're actually reorganizing the control flow such that instead of rescuing *all* exceptions, including `ruby` exceptions, we're now rescuing from specific cases, and any `StandardErrors` will be converted into an `InternalServerError`.

Also on line 79, explain why `exc` instead of `error` is used